library/alloc/src/vec.rs

```diff
@@ -194,6 +194,19 @@ use crate::raw_vec::RawVec;
 /// can be slow. For this reason, it is recommended to use [`Vec::with_capacity`]
 /// whenever possible to specify how big the vector is expected to get.
 ///
+/// In summary, a vector containing 1, 2 with capacity 4 can be visualized as:
+/// ```text
+/// Stack  +--------+--------+--------+
+///        |   ptr  |capacity|   len  |
+///        | 0x0123 |      4 |      2 |
+///        +--------+--------+--------+
+///             |
+///             v
+/// Heap   +--------+--------+--------+--------+
+///        |      1 |      2 |        |        |
+///        +--------+--------+--------+--------+
```

Comment:
maybe add an "uninit" into slots 2 and 3?

Reply:
I wanted to add some empty stuff but I think adding "uninit" would confuse others there is an item "uninit" there.

Reply:
Maybe the item could be linked to the MaybeUninit docs or the rustonomicon.

Reply:
Is it maybeuninit? I am not sure what are the stuff in the empty spaces personally.

Reply:
Well, vec is mostly implemented in terms of `*mut T` which doesn't make any assumptions about things being initialized. But if you wanted to express it as a reference instead of a pointer you'd reach for `&mut MaybeUninit<T>`. But I don't really care that much where it's linked to. I was just suggesting linking *somewhere* that explains issues with uninitialized memory, in case the reader would be confused by "uninit".

Reply:
I added the link.